lib/bitcoin/BitcoinClient.ts

```diff
@@ -313,6 +325,156 @@ export default class BitcoinClient {
+
+    const freezeTransaction = new Transaction()
+                              .from(unspentCoins)
+                              .to(payToScriptAddress, freezeAmountInSatoshis)
+                              .change(this.privateKeyAddress);
+
+    const transactionFee = await this.calculateTransactionFee(freezeTransaction);
+
+    freezeTransaction.fee(transactionFee)
+                     .sign(this.privateKey);
+
+    return freezeTransaction;
+  }
+
+  // @ts-ignore
+  private async createSpendToFreezeBitcoreTransaction (
+    previousFreezeTransaction: Transaction,
+    previousFreezeUntilBlock: number,
+    freezeUntilBlock: number): Promise<Transaction> {
+
+    const freezeScript = this.createFreezeBitcoreScript(freezeUntilBlock);
+    const freezeScriptHash = Script.buildScriptHashOut(freezeScript);
+    const payToScriptAddress = new Address(freezeScriptHash);
+
+    return this.createSpendBitcoreTransactionFromFrozenTransaction(
+      previousFreezeTransaction,
+      previousFreezeUntilBlock,
+      payToScriptAddress);
+  }
+
+  // @ts-ignore
+  private async createSpendToWalletBitcoreTransaction (
+    previousFreezeTransaction: Transaction,
+    previousFreezeUntilBlock: number): Promise<Transaction> {
+
+    return this.createSpendBitcoreTransactionFromFrozenTransaction(
+      previousFreezeTransaction,
+      previousFreezeUntilBlock,
+      this.privateKeyAddress);
+  }
+
+  // @ts-ignore
+  /**
+   * Creates a spend transaction to spend the previously frozen output. The details
+   * on how to create a spend transactions were taken from the BIP65 demo at:
+   * https://github.com/mruddy/bip65-demos/blob/master/freeze.js.
+   *
+   * @param previousFreezeTransaction The previously frozen transaction.
+   * @param previousFreezeUntilBlock The previously frozen transaction's freeze until block.
+   * @param paytoAddress The address where the spend transaction should go to.
+   */
+  private async createSpendBitcoreTransactionFromFrozenTransaction (
+    previousFreezeTransaction: Transaction,
+    previousFreezeUntilBlock: number,
+    paytoAddress: Address): Promise<Transaction> {
+
+    // First create an input from the previous frozen transaction output.
+    const frozenOutputAsInput = this.createBitcoreUnspentOutputFromFrozenTransaction(previousFreezeTransaction, previousFreezeUntilBlock);
+    const previousFreezeAmountInSatoshis = frozenOutputAsInput.satoshis;
+
+    // Now create a spend transaction using the frozen output. Create the transaction with all
+    // inputs and outputs as they are needed to calculate the fee.
+    const spendTransaction = new Transaction()
+                                   .from([frozenOutputAsInput])
+                                   .to(paytoAddress, previousFreezeAmountInSatoshis)
+                                   .lockUntilBlockHeight(previousFreezeUntilBlock);
+
+    const transactionFee = await this.calculateTransactionFee(spendTransaction);

```

Comment:
Is it possible to calculate the fee after script is added to the transaction for a better estimate.

Reply:
The script size does not effect the estimate. We estimate using the number of inputs & outputs.